### PR TITLE
postgresql: fix download URLs

### PR DESCRIPTION
**Pull request by kissg1988**
Maintainer: @dangowrt 
Compile tested: ar71xx, Netgear WNDR4300, openwrt r49984
Run tested: -

Description: downloading the source package fails due to name resolution error and MD5 sum mismatch (seems like providing SHA256 hash does not work for ftp:// links). This patch corrects both issues.

```
--2016-11-09 19:25:47--  http://ftp9.us.postgresql.org/pub/mirrors/postgresql/source/v9.5.4/postgresql-9.5.4.tar.bz2
Resolving ftp9.us.postgresql.org (ftp9.us.postgresql.org)... failed: Name or service not known.
wget: unable to resolve host address 'ftp9.us.postgresql.org'
Download failed.
--2016-11-09 19:25:47--  http://ftp.be.postgresql.org/postgresql/source/v9.5.4/postgresql-9.5.4.tar.bz2
Resolving ftp.be.postgresql.org (ftp.be.postgresql.org)... failed: Name or service not known.
wget: unable to resolve host address 'ftp.be.postgresql.org'
Download failed.
--2016-11-09 19:25:47--  ftp://ftp-archives.postgresql.org/pub/source/v9.5.4/postgresql-9.5.4.tar.bz2
           => '-'
Resolving ftp-archives.postgresql.org (ftp-archives.postgresql.org)... 87.238.57.227, 2a02:c0:301:0:ffff::27
Connecting to ftp-archives.postgresql.org (ftp-archives.postgresql.org)|87.238.57.227|:21... connected.
Logging in as anonymous ... Logged in!
==> SYST ... done.    ==> PWD ... done.
==> TYPE I ... done.  ==> CWD (1) /pub/source/v9.5.4 ... done.
==> SIZE postgresql-9.5.4.tar.bz2 ... 18496299
==> PASV ... done.    ==> RETR postgresql-9.5.4.tar.bz2 ... done.
Length: 18496299 (18M) (unauthoritative)

100%[======================================================================================================================================================================================================================================>] 18,496,299  19.1MB/s   in 0.9s   

2016-11-09 19:25:48 (19.1 MB/s) - written to stdout [18496299]

MD5 sum of the downloaded file does not match (file: ad36fcf624748b8ed67783ad04529f43, requested: cf5e571164ad66028ecd7dd8819e3765470d45bcd440d258b686be7e69c76ed0) - deleting download.
```

### CODE REVIEW

File: libs/postgresql/Makefile [hunk 1]
```diff
@@ -15,10 +15,9 @@ PKG_LICENSE:=PostgreSQL
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
 PKG_SOURCE_URL:=\
-	http://ftp9.us.postgresql.org/pub/mirrors/postgresql/source/v$(PKG_VERSION) \
-	http://ftp.be.postgresql.org/postgresql/source/v$(PKG_VERSION) \
+	https://ftp.postgresql.org/pub/source/v$(PKG_VERSION) \
```
Comment (diizzyy) [inline libs/postgresql/Makefile]: Change the list to the following as Postgresql seems to have dropped specific domain names for mirrors nowdays (drop ftp-archives).
https://ftp.postgresql.org/pub/source/v$(PKG_VERSION)
http://ftp.postgresql.org/pub/source/v$(PKG_VERSION)
ftp://ftp.postgresql.org/pub/source/v$(PKG_VERSION)

File: libs/postgresql/Makefile [hunk 2]
```diff
@@ -15,10 +15,9 @@ PKG_LICENSE:=PostgreSQL
...
 	ftp://ftp-archives.postgresql.org/pub/source/v$(PKG_VERSION)
-PKG_MD5SUM:=cf5e571164ad66028ecd7dd8819e3765470d45bcd440d258b686be7e69c76ed0
+PKG_MD5SUM:=ad36fcf624748b8ed67783ad04529f43
```
Comment (diizzyy) [inline libs/postgresql/Makefile]: Use SHA256, if it doesn't work for ftp (which seems odd) submit bug report upstream

File: libs/postgresql/Makefile (hunk 1 above)
Comment (kissg1988) [inline libs/postgresql/Makefile]: Agreed, will update the list of mirrors.

File: libs/postgresql/Makefile (hunk 2 above)
Comment (kissg1988) [inline libs/postgresql/Makefile]: I have accidentally compiled against chaos calmer which does not support SHA256 hashes. Will restore the hash to its original value.

**Comment by kissg1988** [comment]
Please check and merge my updated changes.

**Comment by diizzyy** [comment]
LGTM

**Comment by hnyman** [comment]
Could somebody please explain the logic of having just and only ftp.postgresql.org there, but with three different protocols? Does it direct the user to different mirrors or is that just bogus?

Note that some of the upstream download mirror systems seem to assign always the user to the same (determined to be nearest?) server, so the built-in mirror logic fails to provide a fallback in reality. Does ftp.postgresql.org lead to different server every time?

If that is not the case, there should preferably be an actual fallback from somewhere.

**Comment by diizzyy** [comment]
I haven't looked into it much but from what I can tell it's a round-robin DNS and for what I know there's no "nice" way of retrying the same URL. The old mirror layout have been scrapped by PostgreSQL and you're directed to ftp.postgresql.org (which is an alias for ftp.mirrors.postgresql.org).

**Pull request merged**